org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/IChangeSerializer.java

```diff
@@ -41,21 +44,51 @@
 @ImplementedBy(ChangeSerializer.class)
 public interface IChangeSerializer {
 
-	<T extends Notifier> void addModification(T context,  IModification<T> modification);
-	   
-	void applyModifications(IAcceptor<IEmfResourceChange>  acceptor);
-	   
+	interface IModification<T extends Notifier> {
+		void modify(T context);
+	}
+
+	<T extends Notifier> void addModification(T context, IModification<T> modification);
+
+	void applyModifications(IAcceptor<IEmfResourceChange> acceptor);
+
 	ITextRegionDiffBuilder getModifyableDocument(Resource resource);
 
+	/**
+	 * @see #setUpdateCrossReferences(boolean)
+	 */
 	boolean isUpdateCrossReferences();
 
+	/**
+	 * @see #setUpdateRelatedFiles(boolean)
+	 */
 	boolean isUpdateRelatedFiles();
 
+	/**
+	 * If set, {@link #addModification(Notifier, IModification)} will report progress and check
+	 * {#{@link IProgressMonitor#isCanceled()} on this monitor object.
+	 */
+	void setProgressMonitor(IProgressMonitor monitor);
+
+	/**
+	 * If set to true (that's the default), cross references affected by changes made in
+	 * {@link #addModification(Notifier, IModification)} will be updated. This involves at least all cross references
+	 * from files for which there is an {@link IModification}. If {@link #isUpdateRelatedFiles()} is true, also all
+	 * other cross references known by {@link IResourceDescriptions} are updated.
+	 * 
+	 * Example: When a modification changes <code>myElement.name</code> from <code>"foo"</code> to <code>"bar"</code>,
+	 * all references pointing to <code>"foo"</code> are updated to <code>"bar"</code>. Yes, that's a rename
+	 * refactoring.
+	 * 
+	 * @see IReferenceUpdater
+	 * @see #setUpdateRelatedFiles(boolean)
+	 */
 	void setUpdateCrossReferences(boolean value);
```

Comment:
What is the usecase of setting one of the properties to false?

Reply:
`setUpdateCrossReferences(false)`:
* repair broken cross references
* better performance when the modification is known to not affect cross references
* have same behavior as the old semantic quickfix / ITextEditComposer implementations
`setUpdateRelatedFiles(false)`
* don't let a broken Xtext index cause wrong results.
* don't risk changing other files except the one that's currently modified.

and probably more...